### PR TITLE
tracy.logger definition can be changed in config

### PR DESCRIPTION
**Pull request by JanTvrdik**
https://forum.nette.org/cs/26317-jak-nastavit-vlastni-logger#p174402

### CODE REVIEW

File: src/Bridges/Nette/TracyExtension.php [hunk 1]
```diff
@@ -76,6 +76,11 @@ public function afterCompile(Nette\PhpGenerator\ClassType $class)
 			}
 		}
 
+		$logger = $builder->getDefinition($this->prefix('logger'));
+		if ($logger->getFactory()->getEntity() !== 'Tracy\Debugger::getLogger') {
+			$initialize->addBody($builder->formatPhp('Tracy\Debugger::setLogger(?);', ['@' . $this->prefix('logger')]));
```
Comment (dg) [inline src/Bridges/Nette/TracyExtension.php]: this is possible too: `[$this->prefix('@logger')]));`

Comment (JanTvrdik) [inline src/Bridges/Nette/TracyExtension.php]: cool, I did not know that

**Comment by kukulich** [comment]
@JanTvrdik Nemělo by to počítat i dalšími formáty `$entity`, které podporuje `Nette\DI\Statement`?

**Comment by dg** [comment]
In aftercompile() are entites normalized (in 2.4)

**Comment by JanTvrdik** [comment]
@kukulich Could you elaborate? 

---

@dg Looking at the code once again, is possible that `$logger->getFactory()` will return `NULL` and `getEntity()` call will fail?

**Comment by dg** [comment]
I think not

**Pull request merged**